client/src/redux/reducers/viewSlice.js

```diff
@@ -0,0 +1,43 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { AppView } from '../../constants/enums'
+
+const viewSlice = createSlice({
+  name: 'view',
+  initialState: {
+    isSidebarOpen: false,
+    appView: AppView.NEW_TRIP,
+    activeTripId: undefined,
+  },
+  reducers: {
+    // payload: AppView (e.g. AppView.NEW_TRIP)
+    setAppView: (state, action) => {
+      state.appView = action.payload
+      state.isSidebarOpen = false
+    },
+    // payload: number (the id of the active trip)
+    setActiveTripId: (state, action) => {
+      state.activeTripId = action.payload
+      state.appView = AppView.TRIP_OVERVIEW
+      state.isSidebarOpen = false
+    },
+    toggleSidebar: (state) => {
+      state.isSidebarOpen = !state.isSidebarOpen
+    },
+    openSidebar: (state) => {
+      state.isSidebarOpen = true
+    },
+    closeSidebar: (state) => {
+      state.isSidebarOpen = false
+    },
+  },
+})
```

Comment:
If these are going to get longer, we should perhaps extract functions out of them and reference the function name.

Reply:
Good idea, done